subprojects/docs/src/samples/java/modules-multi-project-with-integration-tests/README.adoc

```diff
@@ -0,0 +1,28 @@
+NOTE: You can open this sample inside an IDE using the https://www.jetbrains.com/help/idea/gradle.html#gradle_import_project_start[IntelliJ's Gradle import].
+
+This is an link:sample_java_modules_multi_project.html[extension of this sample] that adds blackbox integration tests.
+
+Here, we add an additional source set _integrationTest_ with a `module-info.java`.
+
+```
+src
+└── integrationTest
+    └── java
+        └── module-info.java
+```
+
+We effectively declare a second module that is only used for testing.
+The module is `open`, which means that it allows reflective access to its classes at runtime as required by JUnit's test execution engine.
+
+```
+open module org.gradle.sample.integtest.utilities {
+    requires org.gradle.sample.utilities;
+    requires org.junit.jupiter.api;
+}
+```
+
+NOTE: This sample does **not** work in Eclipse when imported using https://projects.eclipse.org/projects/tools.buildship[Eclipse Buildship].
+This is due to a limitation in Eclipse that does not allow more than one module in one project.
```

Comment:
For what its worth, it may be nice to link the corresponding eclipse bug.

Reply:
Good point. Will do.